src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -305,6 +306,27 @@ public boolean apply(StepDescriptor i) {
         return r;
     }
 
+    /** Return true if we can easily create a nice String for user display from the object */
+    static boolean isAbleToUseToStringForDisplay(@CheckForNull Object o) {
+        return (o != null && (o instanceof Comparable || o instanceof CharSequence)); // Covers our base types
+    }
+
+    /**
+     * Converts user-supplied step arguments to a string for eventual UI use -- override to handle more than a single trivial argument.
+     * Complements {@link #getDisplayName()} in cases where the step type is less meaningful than its arguments (scripts, for example).
+     * <em>Note: this offers a raw value and does not perform escaping on its own.</em>
+     * @param namedArgs List of parameter name-value pairs supplied to the step to instantiate it, as from {@link #uninstantiate(Step)}
+     * @return Formatted string, before escaping, or null if can't be converted easily to a String.
+     */
+    @Nullable
+    public String argumentsToString(@CheckForNull Map<String, Object> namedArgs) {
```

Comment:
I think the method name here is misleading. Probably you would want, say,

```groovy
sh script: 'git rev-parse HEAD', returnStdout: true
```

to return `git rev-parse HEAD`, right? In which case `ShellStep.DescriptorImpl` etc. should be overriding this method to

```java
return namedArgs.get("script");
```

But that is not a representation of the argument**s** in total (which a reader of Javadoc might reasonably expect to look something like `Map.toString`); only a summary of what a user looking at a build log might be most interested in, for purposes of identifying this `Step` among others.

So I think the method name should more clearly reflect that this is a kind of instance-sensitive alternative to `StepDescriptor.getDisplayName`, as the Javadoc hints at.